### PR TITLE
Global refactoring

### PR DESCRIPTION
**Pull request by egorodov**

### CODE REVIEW

File: functions/data_test/profiling.py [hunk 1]
```diff
@@ -21,7 +21,9 @@
 import yaml
 from great_expectations.data_context import BaseDataContext
 
-s3 = boto3.resource("s3")
+
+s3 = boto3.resource("s3", endpoint_url=f"http://{os.environ['S3_HOST']}:4566") if os.environ['ENVIRONMENT'] == 'local' else boto3.resource("s3")
```
Comment (bvolodarskiy) [inline functions/data_test/profiling.py:25]: @a-chumagin what this?

File: examples/basic/main.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+module "data_qa" {
+  source                     = "../../terraform"
+  cloudfront_allowed_subnets = ["213.238.187.134/32"]
+
+  data_test_storage_bucket_name = "dqg-settings"
+  environment                   = "provectus"
+
+  allure_report_image_uri = "024975173233.dkr.ecr.eu-west-2.amazonaws.com/demo-test-ip-allure-report:9b25014b-bf0d-79f4-5a88-9619c4ac42bf"
+  data_test_image_uri     = "024975173233.dkr.ecr.eu-west-2.amazonaws.com/demo-test-ip-fast-data:5b013da8-358b-7d26-e793-7f3140b03f71"
+  push_report_image_uri   = "024975173233.dkr.ecr.eu-west-2.amazonaws.com/demo-test-ip-push-report:ce08702c-5697-a683-a15a-edd1fe1d92dd"
```
Comment (a-chumagin) [inline examples/basic/main.tf:20]: it looks like hardcoded links. is it ok?

**Review by a-chumagin — COMMENTED** [review]
Also, I have two questions 
1. Do we really need to keep terraform plan in repo? [examples/basic/plan](https://github.com/provectus/data-quality-gate/pull/42/files#diff-15ee6f254e2690888f9e059d9f162d1251f7484fe246a9f8863832ab663ef907)
2. Could we download this file instead of keep in reoo? terraform/artifacts/aws-athena-dynamodb-connector.zip
4.

File: .github/workflows/terraform-deploy-data-qa-dev.yml [hunk 3]
```diff
@@ -40,6 +40,50 @@ jobs:
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: eu-west-2
 
+      - name: upload SlackNotification to ECR
+        uses: appleboy/docker-ecr-action@master
+        with:
+          access_key: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          secret_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          registry: 024975173233.dkr.ecr.eu-west-2.amazonaws.com
```
Comment (bvolodarskiy) [inline .github/workflows/terraform-deploy-data-qa-dev.yml:48]: hardcode?

File: examples/basic/main.tf [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  default_tags {
+    tags = {
+      Terraform = true
+    }
+  }
+}
+
+module "data_qa" {
+  source                     = "../../terraform"
+  cloudfront_allowed_subnets = ["213.238.187.134/32"]
```
Comment (bvolodarskiy) [inline examples/basic/main.tf:13]: hardcode?

File: examples/basic/main.tf (hunk 2 above)
Comment (egorodov) [reply examples/basic/main.tf:20]: in basic example yep. let's remove it out

File: .github/workflows/terraform-deploy-data-qa-dev.yml (hunk 3 above)
Comment (egorodov) [reply .github/workflows/terraform-deploy-data-qa-dev.yml:48]: let's move it into the github secrets

File: examples/basic/main.tf (hunk 4 above)
Comment (egorodov) [reply examples/basic/main.tf:13]: Going to use 255.255.255.255/32 as a placeholder

**Review by a-chumagin — APPROVED** [review]
LGTM

**Review by bvolodarskiy — APPROVED** [review]

**Pull request merged**